pkg/edit/vars.go

```diff
@@ -56,6 +55,34 @@ func initVarsAPI(ed *Editor, nb eval.NsBuilder) {
 // ~> echo $b:foo
 // foo
 // ```
+//
+// Because of the above behavior this mechanism is perhaps most useful for adding a module namespace
+// or function to the REPL namespace. For example, you might include this in your
+// `~/.config/elvish/rc.elv` in order to make your Elvish interactive config work on both Windows
+// and UNIX-like systems:
+//
+// ```elvish-transcript
+// use platform
+// if $platform:is-unix {
+//   use unix
+//   edit:add-var unix: $unix:
+// }
+//
+// Similarly you might have functions, such as `$util:ff~` to "find files", that you want to be able
+// to use in the REPL without needing the module prefix (i.e., `ff` rather than `util:ff`) to use
+// the functions interactively. You might add a statement like this to your
+// `~/.config/elvish/rc.elv`:
+//
+// ```elvish-transcript
+// edit:add-var ff~ $util:ff~
+// ```
```

Comment:
Not entirely since the `edit:add-var` is needed if the statement appears in a narrower scope; e.g., an `if` block. Which is the whole point of this command. Also, and more importantly, a hypothetical *lib/util.elv* module which had an `edit:add-var ff~ $ff~` statement makes it impossible to `use util` in a non-interactive context since doing so results in this exception: `Exception: exec: "edit:add-var": executable file not found in $PATH`.